src/Symfony/Component/Uid/Uuid.php

```diff
@@ -130,8 +121,14 @@ final public static function v8(string $uuid): UuidV8
         return new UuidV8($uuid);
     }
 
-    public static function isValid(string $uuid): bool
+    public static function isValid(string $uuid, int $format = self::FORMAT_RFC_4122): bool
```

Comment:
I'm not a big fan of bitfields in options (e.g. `Uuid::FORMAT_BASE_32 | Uuid::FORMAT_RFC_4122`). So, I wonder if we could replace this `int` by a `string|array` (maybe renamed the option too to `$allowedFormats`).

This would be similar to:

* The [formats option](https://symfony.com/doc/current/reference/constraints/CssColor.html#formats) in Css constraint
* The [schemes option](https://symfony.com/doc/current/reference/constraints/CardScheme.html#schemes) in CardScheme constraint

Reply:
I like bit-fields because it reads well, it is quick and easy to test a flag, but also because it offers nice auto-completion and static analysis thanks to `@param int-mask-of`.

Those seems a bit more difficult with a `string|array`. I'm curious to understand why you're not a big fan of bit-field and why'd you prefer an array of options? :slightly_smiling_face:

Reply:
A bit-field is more appropriate to me also.

Note that adding an argument is a BC break, we should comment it and use func_get_arg instead

Reply:
Oh yes, I got confused with private methods. It's updated now